components/director/internal/domain/label/query_test.go

```diff
@@ -0,0 +1,97 @@
+package label
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+)
+
+func Test_FilterQuery(t *testing.T) {
```

Comment:
Such query is not going to work:
```
SELECT "runtime_id" FROM "public"."labels" WHERE "tenant_id" = 'aa5eac41-c2a4-4ead-95b6-5574505e81c5' AND "key" = 'Foo' AND "key" = 'Bar'
```

I would expect that the following query will be created:
```
select l1.app_id from labels l1 join labels l2 on l1.app_id =l2.app_id where l1.key='foo' and l2.key='bar';
```

but I am wondering about the performance of such query...

Reply:
I'm not a fan of this approach as when we will have more labels to filter then the join will grow and grow.

Reply:
The alternative to the join queries will be the intersect.

JOIN:
`select l1.runtime_id from labels l1 join labels l2 on l2.runtime_id = l1.runtime_id and l2.key = 'label1' join labels l3 on l3.runtime_id = l1.runtime_id and l3.key = 'label3' where l1.key = 'label1';`

INTERSECT
`select runtime_id from labels where key = 'label1' intersect select runtime_id from labels where key = 'label2' intersect select runtime_id from labels where key = 'label3';`